### PR TITLE
add test

### PR DESCRIPTION
**Pull request by onsd**
## Why
ref. #15 
I'd like to be able to Update resources, but it doesn't have any test so I don't know if I break it.

## What
- Write some test 
	- do nothing if there are any resources
	- do nothing if there is only deployment-copy
	- create copied deployment if there are deployment and related deployment-copy
	- update copied deployment if there are deployment, related deployment-copy and copied-deployment
		- currently, it can't update these copied-deployment so **it shows current behavior**

### CODE REVIEW

**Comment by potsbo** [comment]
Would you rewrite the description in English?

File: controllers/deploymentcopy_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	"k8s.io/apimachinery/pkg/types"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/wantedly/deployment-duplicator/controllers"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	ut "github.com/wantedly/deployment-duplicator/controllers/testing"
+	"testing"
```
Comment (potsbo) [inline controllers/deploymentcopy_controller_test.go:16]: It looks like this file needs `gofmt`

File: controllers/deploymentcopy_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		},
+		{
+			name:        "only deployment copy",
+			explanation: "do nothing because there's no deployment",
+			initialState: []runtime.Object{
+				ut.GenDeploymentCopy("some-deployment-copy", "some-deployment", ut.AddTargetContainer("some-container", "some-image-tag")),
+			},
+		},
+		{
+			name:        "one deployment one deployment copy",
```
Comment (potsbo) [inline controllers/deploymentcopy_controller_test.go:53]:
```suggestion
			name:        "one deployment and one deployment copy",
```

File: controllers/testing/helpers.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	ddv1beta1 "github.com/wantedly/deployment-duplicator/api/v1beta1"
+	"gopkg.in/yaml.v2"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"testing"
+
+	"github.com/bradleyjkemp/cupaloy/v2"
+	"github.com/stuart-warren/yamlfmt"
```
Comment (potsbo) [inline controllers/testing/helpers.go:16]: This file also needs `gofmt`

File: controllers/testing/helpers.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+		opt(d)
+	}
+	return d
+}
+
+func AddContainer(name, image string) deploymentOption {
+	return func(d *appsv1.Deployment) {
+		if d.Spec.Template.Spec.Containers == nil {
+			d.Spec.Template.Spec.Containers = []v1.Container{}
+		}
```
Comment (potsbo) [inline controllers/testing/helpers.go:56]: I don't think we need this block.
`append` can take `nil` for the first arg.

https://play.golang.org/p/4JeD-cP78AV

Comment (onsd) [reply controllers/testing/helpers.go]: I didn't know it, but it works for me too 😄

**Comment by onsd** [comment]
Add some commits
- run `go fmt`
- remove unnecessary blocks

**Review by potsbo — APPROVED** [review]
LGTM

**Pull request merged**